source/client/factories_impl.cc

```diff
@@ -118,16 +118,24 @@ void RequestSourceFactoryImpl::setRequestHeader(Envoy::Http::HeaderMap& header,
 }
 
 RequestSourcePtr RequestSourceFactoryImpl::create() const {
-  // Note: we assume a valid uri.
-  // Also, we can't resolve, but we do not need that.
-  UriImpl uri(options_.uri());
   Envoy::Http::HeaderMapPtr header = std::make_unique<Envoy::Http::HeaderMapImpl>();
+  if (options_.uri().has_value()) {
+    // Note: we assume a valid uri.
+    // Also, we can't resolve, but we do not need that.
```

Comment:
The comment as it stands begs a question - why can't we resolve? This is probably a result of the review discussion that happened here, but can we improve (or remove) the comment?

Reply:
Updated comment based on my understanding.